Replace all Prometheus datasource UIDs of the Grafana Dashboard with the variable `${DS_PROMETHEUS}` and remove `__inputs`

MichaelSasser opened the pull request:
Fixes: #16405 

### Pull Request Checklist

<!-- Please read https://matrix-org.github.io/synapse/latest/development/contributing_guide.html before submitting your pull request -->

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [ ] <del>[Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))</del> **Not applicable**

### What was changed

I replaced the `000000001` and `$datasource` Prometheus datasource UIDs `${DS_PROMETHEUS}` using regular expressions. Then I added the label "Datasource" to the `DS_PROMETHEUS` variable and removed the `__inputs` section, so Grafana doesn't treat the variable as a placeholder that it replaces on Import.

No errors on import, panels show data. Tested with Mimir 2.9.1 and a fresh Grafana 9.5.13 with Angular support enabled (due to the deprecated Angular-based panels).

Signed-off-by: Michael Sasser <michael@michaelsasser.org>

DMRobertson reviewed (approved):
> No errors on import, panels show data. Tested with Mimir 2.9.1 and a fresh Grafana 9.5.13 with Angular support enabled (due to the deprecated Angular-based panels).

Super, thanks for confirming (and more generally for your diligence in fixing this).

MichaelSasser commented:
There is a [preview feature toggle](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/feature-toggles/#preview-feature-toggles) called `autoMigrateOldPanels` (used like `GF_FEATURE_TOGGLES_ENABLE=autoMigrateOldPanels` or `GF_FEATURE_TOGGLES_ENABLE=autoMigrateOldPanels anotherFeatureAsExampleThatDoesNotExist [...]` for multiple feature) that automatically migrates the deprecated Angular panels.

> Migrate old angular panels to supported versions (graph, table-old, worldmap, etc)

I have enabled it with Grafana 10.x when I moved the LGTM stack to a new server a few weeks ago. The feature does work semi-well. Most of the panels were migrated on import, but some did not. I haven't looked too much into it, but I have the feeling that it has to do with panels where the series does not exist on import yet. If I recall it correctly, I imported the alerts first, then the dashboards and later on, the recording rules because I forgot them for a minute. A large number of the panels that were not migrated had either used recordings or variables based on labels and filters that I changed in Mimir but not yet in the dashboard.

When I find some time, I'll try it again, do the rest manually if needed and hand in a PR.
